src/main/host/memory_manager.rs

```diff
@@ -953,14 +1002,19 @@ impl MemoryManager {
         Ok(())
     }
 
-    // Extend the portion of the stack that we've mapped downward to include `src`.  This is
-    // carefuly designed *not* to invalidate any outstanding borrowed references or pointers, since
-    // otherwise a caller trying to marshall multiple syscall arguments might invalidate the first
-    // argument when marshalling the second.
+    // Extend the portion of the stack that we've mapped downward to include `src`.
+    //
+    // This is carefuly designed *not* to invalidate any outstanding borrowed references or
+    // pointers, since otherwise a caller trying to marshall multiple syscall arguments might
+    // invalidate the first argument when marshalling the second. (While the Rust API currently
+    // prevents there from being any outstanding references, the C API does not).
```

Comment:
Make these `///` so that they show up in the docs? (And for other functions in this file.)

Reply:
Does it make sense to use `///` for non-public methods? I suppose it doesn't hurt...

Reply:
I think it makes sense, since if you're developing the library itself and not just using it in another project, you may run `cargo doc --document-private-items` which will include those functions. And rust seems to use `///` for non-public methods as well: https://doc.rust-lang.org/src/alloc/vec.rs.html#2978

Reply:
SG. Thanks!